src/topology/alexandroff.lean

```diff
@@ -0,0 +1,370 @@
+-/
+
+/-- For any topological space `X`, its one point compactification is a compact space. -/
+instance : compact_space (alexandroff X) :=
+{ compact_univ :=
+  begin
+    refine is_compact_iff_ultrafilter_le_nhds.2 (λ f hf, _), clear hf,
+    by_cases hf : (f : filter (alexandroff X)) ≤ 𝓝 ∞,
+    { exact ⟨∞, mem_univ _, hf⟩ },
+    { simp only [ultrafilter_le_nhds_infty, not_forall, not_not] at hf,
+      rcases hf with ⟨s, h₁, h₂, hsf⟩,
+      have hf : range (coe : X → alexandroff X) ∈ f,
+        from mem_sets_of_superset hsf (image_subset_range _ _),
+      have hsf' : s ∈ f.comap coe_injective hf, from (f.mem_comap _ _).2 hsf,
+      rcases h₂.ultrafilter_le_nhds _ (le_principal_iff.2 hsf') with ⟨a, has, hle⟩,
+      rw [ultrafilter.coe_comap, ← comap_coe_nhds, comap_le_comap_iff hf] at hle,
+      exact ⟨a, mem_univ _, hle⟩ }
+  end }
+
+/-- The one point compactification of a `t0_space` space is a `t0_space`. -/
+instance [t0_space X] : t0_space (alexandroff X) :=
+begin
+  refine ⟨λ x y hxy, _⟩,
+  induction x using alexandroff.rec; induction y using alexandroff.rec,
+  { exact (hxy rfl).elim },
+  { use {∞}ᶜ, simp [is_closed_infty] },
+  { use {∞}ᶜ, simp [is_closed_infty] },
+  { rcases t0_space.t0 x y (mt coe_eq_coe.mpr hxy) with ⟨U, hUo, hU⟩,
+    refine ⟨coe '' U, is_open_image_coe.2 hUo, _⟩,
+    simpa }
+end
+
+/-- The one point compactification of a `t1_space` space is a `t1_space`. -/
+instance [t1_space X] : t1_space (alexandroff X) :=
+{ t1 := λ z,
+  begin
+    induction z using alexandroff.rec,
+    { exact is_closed_infty },
+    { simp only [← image_singleton, is_closed_image_coe],
+      exact ⟨is_closed_singleton, is_compact_singleton⟩ }
+  end }
+
+/-- The one point compactification of a locally compact Hausdorff space is a normal (hence,
+Hausdorff and regular) topological space. -/
+instance [locally_compact_space X] [t2_space X] : normal_space (alexandroff X) :=
+begin
+  have key : ∀ z : X,
+    ∃ u v : set (alexandroff X), is_open u ∧ is_open v ∧ ↑z ∈ u ∧ ∞ ∈ v ∧ u ∩ v = ∅,
+  { intro z,
+    rcases exists_open_with_compact_closure z with ⟨u, hu, huy', Hu⟩,
+    refine ⟨coe '' u, (coe '' closure u)ᶜ, is_open_image_coe.2 hu,
+      is_open_compl_image_coe.2 ⟨is_closed_closure, Hu⟩, mem_image_of_mem _ huy',
+      mem_compl infty_not_mem_image_coe, _⟩,
+    rw [← subset_compl_iff_disjoint, compl_compl],
+    exact image_subset _ subset_closure },
+  refine @normal_of_compact_t2 _ _ _ ⟨λ x y hxy, _⟩,
+  induction x using alexandroff.rec; induction y using alexandroff.rec,
+  { exact (hxy rfl).elim },
+  { rcases key y with ⟨u, v, hu, hv, hxu, hyv, huv⟩,
+    exact ⟨v, u, hv, hu, hyv, hxu, (inter_comm u v) ▸ huv⟩ },
+  { exact key x },
+  { exact separated_by_open_embedding open_embedding_coe (mt coe_eq_coe.mpr hxy) }
+end
+
+/-- If `X` is not a compact space, then `alexandroff X` is a connected space. -/
+lemma connected_space_alexandroff [preconnected_space X] (h : ¬ is_compact (univ : set X)) :
```

Comment:
Could you mention "This statement cannot be an `instance` because it needs a non-class argument `¬is_compact (univ : set X)`." here, too?